IMA-WorldHealth/bhima
# Add DROPs before all SQL CREATE PROCEDURE/FUNCTION definitions

Update the Bhima SQL procedure files to make sure they can be reloaded without causing errors or warnings.  This is useful in at least two cases to avoid problems when reloading procedure files: (1) while creating new procedures/functions or modifying old procedures/functions, and (2) when  dealing with migrations, sometimes it is necessary to reload certain procedure files to update procedure/function definitions.

**Adds** 
- DROP PROCEDURE IF EXISTS before all CREATE PROCEDURE ...
- DROP FUNCTION IF EXISTS before all CREATE FUNCTION ...

Closes https://github.com/IMA-WorldHealth/bhima/issues/6036

**TESTING**
- Suggest testing with bhima_test
- Run:  yarn; yarn build
- Make sure unit and integration tests run okay
- Download this script: [stest.txt](https://github.com/IMA-WorldHealth/bhima/files/7371767/stest.txt)
- Put stest.txt into the top level of your Bhima installation
- Edit stest.txt to add in bhima_test db password
- Execute it:  bash stest.txt
- You should see complaints about using the db password on the command line, BUT NO OTHER errors or warnings
- Run the tests again.  They should pass like before
-
